bench-exchange/src/bench.rs

```diff
@@ -92,14 +104,44 @@ where
     );
 
     let accounts_in_groups = batch_size * account_groups;
-    let exit_signal = Arc::new(AtomicBool::new(false));
-    let clients: Vec<_> = clients.into_iter().map(Arc::new).collect();
-    let client = clients[0].as_ref();
-
-    const NUM_KEYPAIR_GROUPS: u64 = 4;
+    const NUM_KEYPAIR_GROUPS: u64 = 2;
     let total_keys = accounts_in_groups as u64 * NUM_KEYPAIR_GROUPS;
-    info!("Generating {:?} keys", total_keys);
-    let mut keypairs = generate_keypairs(total_keys);
+
+    if write_to_client_file {
```

Comment:
This could be moved after the below if/else and use the keypairs read/generated there, thus saving a call to generate keypairs?

Reply:
This if clause does a return at the end. So keypairs are not created twice. I see that it'll still function the same if we were to move it after if/else block. I feel that the code will become little harder to follow, though. It'll rely on the fact that `write_to_client_file` and `read_from_client_file` are mutually exclusive.

Reply:
Agreed